### DISCUSSION TITLE
Reading the pipeline variables inside the handlebars template

### DISCUSSION THREAD

**Discussion by mlop3s** [rfennell/AzurePipelines#983] [Q&A] [answered]
Hi,

is there a way to read the env variables from within the template.
Like 
## Release notes for $env:GITVERSION_SEMVER
**Build Number**: {{buildDetails.id}}

Perhaps process.env.env:GITVERSION_SEMVER ?

Thanks.
Marco

**Comment by rfennell** [comment, marked as answer]
No there is way to read environmental variables on the template. However, if it is semver you want why not set the build name using the GitVersion tasks. 

This is the way I do it, I use the following at the top of my build

`name: $(Build.DefinitionName)_$(GitVersion.FullSemVer)`

And then can use this in the templates as

` Build {{buildDetails.buildNumber}}`

**Reply by mlop3s** [reply]
Hi,
yes, this would be a way to do it. I decided to go for an extension:
```
    getProductVersion() {
        return process.env.GITVERSION_SEMVER;
    } 
```
Which actually could be expanded to get any variable. Would be nice to have it as default
Something like builddetails.env.GITVERSION_SEMVER.

Anyway, it looks nice now

# Notes for build {{getProductVersion}}

Cheers,
Marco

**Reply by rfennell** [reply]
Thats just the sort of reason the extension model exists. Everyone's needs are different.  Glad you git sirted

**Reply by markbangert** [reply]
getProductVersion() {
        return process.env.GITVERSION_SEMVER;
    } 

Just wanted to add that this syntax will only work on windows agents.

**Reply by ejwill** [reply]
could use getVariable() or getVariables() from the azure-pipelines-task-lib then you just need to pipe the data to the template 

https://github.com/microsoft/azure-pipelines-task-lib/blob/master/node/docs/azure-pipelines-task-lib.md#taskgetvariable-

**Reply by huwbristowaltiaintel** [reply]
Thanks for the responses. I think it is this bit I am struggling with "pipe the data to the template" please could you show syntax for that?
